### PR TITLE
STORM-2499: Add Serialization plugin for EventHub System Properties

### PR DESCRIPTION
**Pull request by rban1**
This PR adds a serialization scheme for eventhub systemproperties. There is also a bug fix in an existing serialization scheme which violates the schema setting.

### CODE REVIEW

File: external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/BinaryEventDataSystemPropertiesScheme.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * An Event Data Scheme which deserializes message payload into the raw bytes.
+ *
+ * The resulting tuple would contain two items, the first being the message
+ * bytes, and the second a map of properties that include System Metadata
+ * properties. System Properties in eventhub exposes offset, sequencenumber
+ * and EnqueueTime
+ */
+public class BinaryEventDataSystemPropertiesScheme implements IEventDataScheme{
```
Comment (SreeramGarlapati) [inline external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/BinaryEventDataSystemPropertiesScheme.java]: >BinaryEventDataSystemPropertiesScheme [](start = 13, length = 37)

add a with in the name.. EventDataWithSys....

File: external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/BinaryEventDataSystemPropertiesScheme.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An Event Data Scheme which deserializes message payload into the raw bytes.
+ *
+ * The resulting tuple would contain two items, the first being the message
+ * bytes, and the second a map of properties that include System Metadata
```
Comment (SreeramGarlapati) [inline external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/BinaryEventDataSystemPropertiesScheme.java]: >properties [](start = 34, length = 10)

Indicate explicitly that EventData.Properties are not included
or I would actually recommend to go ahead and include eventdata.getProperties() also in the deserialized output.

File: external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/FieldConstants.java [hunk 3]
```diff
@@ -23,5 +23,6 @@
   public static final String Offset = "offset";
   public static final String Message = "message";
   public static final String META_DATA = "metadata";
+  public static final String SYSTEM_META_DATA = "systemmetadata";
```
Comment (SreeramGarlapati) [inline external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/FieldConstants.java]: >SYSTEM_META_DATA [](start = 29, length = 16)

Too generic; in general being more specific is recommended - feel free to use something like eventdata_system_properties

File: external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/BinaryEventDataSystemPropertiesScheme.java (hunk 2 above)
Comment (rban1) [inline external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/BinaryEventDataSystemPropertiesScheme.java]: Will make this serialization scheme have both Properties and SystemProperties

File: external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/FieldConstants.java (hunk 3 above)
Comment (rban1) [inline external/storm-eventhubs/src/main/java/org/apache/storm/eventhubs/spout/FieldConstants.java]: will change it to eventdata_system_properties

**Comment by harshach** [comment]
@rban1 looks good. Can you add the new config to README and also squash the commits into 1

**Comment by rban1** [comment]
@harshach Thanks for the review. I have squashed commits into 1 and added a section in the README

**Comment by rban1** [comment]
@harshach Can you please merge it if the changes look good

**Pull request merged**